src/axom/primal/tests/primal_solid_angle.cpp

```diff
@@ -478,6 +478,195 @@ TEST(primal_solid_angle, selfintersecting_quadrilateral)
   }
 }
 
+//------------------------------------------------------------------------------
+TEST(primal_solid_angle, planar_bezierpatch)
+{
+  using Point3D = primal::Point<double, 3>;
+  using Vector3D = primal::Vector<double, 3>;
+  using Polygon = primal::Polygon<double, 3>;
+  using BezierPatch = primal::BezierPatch<double>;
+
+  // Define normal vector for the quadrilateral
+  Vector3D v1 = Vector3D({0.0, 1.0, 2.0}).unitVector();
+  Vector3D v2 = Vector3D({2.0, -1.0, 0.5}).unitVector();
+
+  Polygon quad(4);
+
+  double angles[5] = {1.0, 1.5, 2.5, 3.0};
+  // Add vertices to quadrilateral
+  for(int i = 0; i < 4; ++i)
+  {
+    quad.addVertex(Point3D {cos(angles[i]) * v1[0] + sin(angles[i]) * v2[0],
+                            cos(angles[i]) * v1[1] + sin(angles[i]) * v2[1],
+                            cos(angles[i]) * v1[2] + sin(angles[i]) * v2[2]});
+  }
+
+  // Construct a first order Bezier patch out of the same vertices
+  Point3D controlPoints[4] = {quad[1], quad[0], quad[2], quad[3]};
+  BezierPatch quad_patch(controlPoints, 1, 1);
+
+  Point3D queries[5] = {
+    Point3D {0.0, 4.0, 1.0},
+    Point3D {-1.0, 2.0, 2.0},
+    Point3D {0.0, -5.0, 3.0},
+    Point3D {0.0, 0.0, 4.0},
+    Point3D {3.0, 2.0, 5.0},
+  };
+
+  // Should be equal with both kinds of primitive
+  for(int n = 0; n < 5; ++n)
+  {
+    EXPECT_NEAR(winding_number(queries[n], quad),
+                winding_number(queries[n], quad_patch),
+                1e-10);
+  }
+
+  // Should be equal if we don't revert to polygon base case
+  const double quad_tol = 1e-10;
+  const double edge_tol = 1e-10;
+  const double EPS = 0;
+  for(int n = 0; n < 5; ++n)
+  {
+    EXPECT_NEAR(winding_number(queries[n], quad_patch),
+                winding_number(queries[n], quad_patch, quad_tol, edge_tol, EPS),
+                1e-10);
+  }
+}
+
+//------------------------------------------------------------------------------
+TEST(primal_integral, bezierpatch_sphere)
+{
+  using Point3D = primal::Point<double, 3>;
+  using Vector3D = primal::Vector<double, 3>;
+  using BPatch = primal::BezierPatch<double>;
+
+  double rt2 = sqrt(2), rt3 = sqrt(3), rt6 = sqrt(6);
+
+  // Define 6 rational bezier patches that define the surface of a sphere
+  // clang-format off
+  Point3D node_data[25] = {
```

Comment:
Suggestion: It might be cleaner/clearer to make this an `axom::Array<Point3D, 2>`, assuming that's supported
(and if not, we should fix it)

Reply:
Far as I can tell, `axom::Array<T, 2>` can't support nested initializer lists. I have changed it to use 1D `axom::Array`s though, which is a little nicer.

Reply:
@publixsubfan (or others) -- do we have a nice way of initializing 2D arrays, e.g. w/ nested initializer lists?

I suppose it could be a bit awkward since our 2D arrays are not ragged...